client/my-sites/stats/geochart/index.jsx

```diff
@@ -50,75 +64,99 @@ export default React.createClass( {
+	drawRegionsMap = () => {
+		if ( this.refs && this.refs.chart ) {
 			this.visualizationsLoaded = true;
 			this.visualization = new window.google.visualization.GeoChart( this.refs.chart );
 			window.google.visualization.events.addListener( this.visualization, 'regionClick', this.recordEvent );
 
 			this.drawData();
 		}
-	},
+	}
 
-	resize() {
+	resize = () => {
 		if ( this.visualizationsLoaded ) {
 			this.drawData();
 		}
-	},
-
-	drawData() {
-		const regionCodes = [];
-		const { data, dataList } = this.props;
-
-		if ( data.length > 1 ) {
-			const chartData = window.google.visualization.arrayToDataTable( data );
-			const node = this.refs.chart;
-			const width = node.clientWidth;
-
-			const options = {
-				width: 100 + '%',
-				height: width <= 480 ? '238' : '480',
-				keepAspectRatio: true,
-				enableRegionInteractivity: true,
-				region: 'world',
-				colorAxis: { colors: [ '#FFF088', '#F34605' ] }
-			};
-
-			dataList.response.data.map( function( country ) {
-				if ( regionCodes.indexOf( country.region ) === -1 ) {
-					regionCodes.push( country.region );
-				}
-			} );
-
-			if ( 1 === regionCodes.length ) {
-				options.region = regionCodes[ 0 ];
-			}
-
-			this.visualization.draw( chartData, options );
+	}
+
+	drawData = () => {
+		const { data } = this.props;
+
+		if ( ! data || ! data.length ) {
+			return;
+		}
+
+		const mapData = map( data, ( country ) => {
+			return [ country.label, country.value ];
+		} );
+		mapData.unshift( [
+			this.props.translate( 'Country' ).toString(),
+			this.props.translate( 'Views' ).toString()
+		] );
+
+		const chartData = window.google.visualization.arrayToDataTable( mapData );
+		const node = this.refs.chart;
+		const width = node.clientWidth;
+
+		const options = {
+			width: 100 + '%',
+			height: width <= 480 ? '238' : '480',
+			keepAspectRatio: true,
+			enableRegionInteractivity: true,
+			region: 'world',
+			colorAxis: { colors: [ '#FFF088', '#F34605' ] }
+		};
+
+		const regions = uniq( map( data, 'region' ) );
+
+		if ( 1 === regions.length ) {
+			options.region = regions[ 0 ];
 		}
-	},
 
-	loadVisualizations() {
+		this.visualization.draw( chartData, options );
+	}
+
+	loadVisualizations = () => {
 		// If google is already in the DOM, don't load it again.
 		if ( window.google ) {
 			window.google.load( 'visualization', '1', { packages: [ 'geochart' ], callback: this.drawRegionsMap } );
 			clearTimeout( this.timer );
 		} else {
 			this.tick();
 		}
-	},
-
-	tick() {
-		this.timer = setTimeout( this.loadVisualizations, 1000 );
-	},
+	}
 
 	render() {
-		return <div ref="chart" className="stats-geochart" />;
+		const { siteId, statType, query, data } = this.props;
+		const isLoading = ! data;
+		const classes = classNames( 'stats-geochart', {
+			'is-loading': isLoading
+		} );
+		return (
+			<div>
+				<div ref="chart" className={ classes } />
+				{ siteId && statType && <QuerySiteStats statType={ statType } siteId={ siteId } query={ query } /> }
```

Comment:
Both are required, but `statType` will never be `null`.  Habits.  I'll remove the `statType` check.